crates/walnut/src/verification.rs

```
@@ -0,0 +1,202 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use anyhow::{anyhow, Context, Result};
+use console::{pad_str, Alignment, Style, StyledObject};
+use dojo_world::metadata::get_default_namespace_from_ws;
+use dojo_world::migration::strategy::MigrationStrategy;
+use futures::future::join_all;
+use reqwest::StatusCode;
+use scarb::core::Workspace;
+use serde::Serialize;
+use serde_json::Value;
+use walkdir::WalkDir;
+
+/// Verifies all classes declared during migration.
+/// Only supported on hosted networks (non-localhost).
+pub async fn walnut_verify_migration_strategy(
+    ws: &Workspace<'_>,
+    rpc_url: String,
+    migration_strategy: &MigrationStrategy,
+) -> Result<()> {
+    let ui = ws.config().ui();
+    // Check if rpc_url is localhost
+    if rpc_url.contains("localhost") || rpc_url.contains("127.0.0.1") {
+        ui.print(" ");
+        ui.print("Verifying classes with Walnut is only supported on hosted networks.");
```

We should warn here, as we fail silently.